src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -39,90 +39,216 @@
+import io.cryostat.net.web.WebServer;
 import io.cryostat.platform.PlatformClient;
+import io.cryostat.rules.ArchivePathException;
+import io.cryostat.rules.ArchivedRecordingInfo;
 import io.cryostat.util.URIUtil;
 
+import org.apache.commons.codec.binary.Base32;
+
 public class RecordingArchiveHelper {
 
     private final TargetConnectionManager targetConnectionManager;
     private final FileSystem fs;
+    private final Provider<WebServer> webServerProvider;
+    private final Logger logger;
     private final Path recordingsPath;
     private final Clock clock;
     private final PlatformClient platformClient;
     private final ReportService reportService;
 
     RecordingArchiveHelper(
             FileSystem fs,
+            Provider<WebServer> webServerProvider,
+            Logger logger,
             @Named(MainModule.RECORDINGS_PATH) Path recordingsPath,
             TargetConnectionManager targetConnectionManager,
             Clock clock,
             PlatformClient platformClient,
             ReportService reportService) {
         this.fs = fs;
+        this.webServerProvider = webServerProvider;
+        this.logger = logger;
         this.recordingsPath = recordingsPath;
         this.targetConnectionManager = targetConnectionManager;
         this.clock = clock;
         this.platformClient = platformClient;
         this.reportService = reportService;
     }
 
-    public String saveRecording(ConnectionDescriptor connectionDescriptor, String recordingName)
-            throws Exception {
+    public Future<String> saveRecording(
+            ConnectionDescriptor connectionDescriptor, String recordingName) throws Exception {
+
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        try {
+            String saveName =
+                    targetConnectionManager.executeConnectedTask(
+                            connectionDescriptor,
+                            connection -> {
+                                Optional<IRecordingDescriptor> descriptor =
+                                        this.getDescriptorByName(connection, recordingName);
+
+                                if (descriptor.isPresent()) {
+                                    return writeRecordingToDestination(
+                                            connection, descriptor.get());
+                                } else {
+                                    throw new RecordingNotFoundException(recordingName);
+                                }
+                            });
+            future.complete(saveName);
+        } catch (RecordingNotFoundException e) {
+            future.completeExceptionally(e);
+        }
+
+        return future;
+    }
+
+    public Future<Void> deleteRecording(String recordingName) throws Exception {
+
+        CompletableFuture<Void> future = new CompletableFuture<>();
+
+        try {
+            Path archivedRecording = getRecordingPath(recordingName).get();
```

Comment:
Maybe the returned `Future` could relay this Path to the caller as well - they may or may not need it, but it's a piece of potentially useful information.